src/Amalgam/AssetManager.cpp

```diff
@@ -241,6 +322,20 @@ Entity *AssetManager::LoadEntityFromResource(AssetParameters &asset_params, bool
 	Entity *new_entity = new Entity();
 	new_entity->SetRandomState(default_random_seed, true);
 
+	if(asset_params.executeOnLoad && asset_params.transactional)
+	{
+		if(!LoadResourceViaTransactionalExecution(asset_params, new_entity, calling_interpreter, status))
+		{
+			delete new_entity;
```

Comment:
I'm not sure if this is a path worth optimizing, but there is an opportunity to avoid allocation here if we push allocation of new_entity into LoadResouceViaTransactionalExecution, and it just doesn't allocate it sometimes. Rather than allocated and then deleting.

Reply:
I guess it would make the code too awkward for the benefit.